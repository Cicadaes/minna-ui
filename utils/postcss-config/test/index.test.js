/** @jest-environment node */

'use strict';

const fs = require('fs');
const path = require('path');
const { promisify } = require('util');
const postcss = require('postcss');
const postcssConfig = require('../index.js');

const readFile = promisify(fs.readFile);
const jestConfigPath = require.resolve('@minna-ui/jest-config');
const fixturesPath = path.join(path.dirname(jestConfigPath), 'fixtures');
const sourceCssPath = path.join(fixturesPath, 'import.css');
const sourceCssMixinPath = path.join(fixturesPath, 'mixin.css');
const mixinsPath = path.join(fixturesPath, 'css-mixins');

const options = {
  from: sourceCssPath,
  map: {
    inline: false,
    annotation: path.basename(sourceCssPath),
  },
};

let sourceCss = '';
let sourceCssMixin = '';

beforeAll(async () => {
  [sourceCss, sourceCssMixin] = await Promise.all([
    readFile(sourceCssPath, 'utf8'),
    readFile(sourceCssMixinPath, 'utf8'),
  ]);
});

describe('PostCSS config', () => {
  it('compiles valid CSS', async () => {
    const result = await postcss(postcssConfig())
      .process(sourceCss, options);
    expect(result.processor.plugins).not.toHaveLength(0);
    expect(result.opts.from).toBeDefined();
    expect(result.map).toBeDefined();
    expect(result.css).toMatchSnapshot();
  });

  it('compiles CSS with custom mixin', async () => {
    const result = await postcss(postcssConfig({ mixinsPath }))
      .process(sourceCssMixin, { from: sourceCssMixinPath });
    expect(result.css).toMatch('.target::after');
    expect(result.css).toMatchSnapshot();
  });
});
