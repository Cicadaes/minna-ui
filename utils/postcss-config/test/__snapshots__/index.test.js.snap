// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCSS config compiles CSS with custom mixin 1`] = `
".target::after {
    display: block;
    clear: both;
    content: \\"\\";
  }
"
`;

exports[`PostCSS config compiles valid CSS 1`] = `
":root { /* red */
}
body {
  margin: 0;
}
/* stylelint-disable selector-max-id */
#target {
  color: hsl(0, 100%, 50%);
}
#target .focus,
  #target:focus {
    background: hsl(0, 0%, 0%);
  }
.wrapper + #target {
    background: hsl(0, 0%, 13%);
  }
/* stylelint-enable selector-max-id */

/*# sourceMappingURL=import.css */"
`;
